modules/net-vpc-factory/README.md

```diff
@@ -0,0 +1,837 @@
+
+### Cross-Referencing Resources
+
+A key capability of this factory is establishing relationships between resources defined across different VPCs or projects. This is achieved through cross-referencing using logical keys within the YAML configuration. These keys strictly follow the pattern `project_key/vpc_key/resource_key` for resources like VPCs, VPN gateways, routers, or NCC hubs/groups. During Terraform execution, the factory code translates these logical keys into the actual resource IDs or self-links required by the underlying modules. This  mechanism avoids hardcoding resource IDs and enables the declarative definition of complex topologies. It's extensively used in configurations for:
+
+VPC Peering: Specifying the peer_network (e.g., `peer_network: net-land-01/hub`).
+
+DNS Peering/Forwarding: Defining peer_network or client_networks for DNS zones (e.g., `client_networks: [net-dev-01/dev-spoke]`).
+
+GCP-to-GCP HA VPN: Referencing the peer HA VPN gateway using its logical key (e.g., `peer_gateways.default.gcp: net-dev-01/dev-spoke/to-hub`).
+
+Network Connectivity Center (NCC): Linking spokes to hubs (`ncc_config.hub: net-land-01/hub`) and groups (`ncc_config.group: net-land-01/hub/default`), or auto-accepting spokes in hub groups (`auto_accept: [net-prod-01, net-dev-01]`).
+
+## Projects
+
+The `project_config` block within each YAML file implements a large subset of the [project module](../project/) variable interface, which allows amongst the rest for project creation or reuse, services enablement and IAM/Organization policies definition.
+
+Below a valid YAML file which simply creates a project, enables a minimal set of services, configures the project as a host project, adds an authoritative
+role binding and sets an Organization Policy at the project level:
+
+```hcl
+module "net-vpc-factory" {
+  source           = "./fabric/modules/net-vpc-factory"
+  billing_account  = "123456-789012-345678"
+  parent_id        = "folders/123456789012"
+  prefix           = "myprefix"
+  factories_config = { vpcs = "recipes/examples" }
+}
+# tftest files=project-config
+```
+
+```yaml
+project_config:
+  name: project-config
+  services:
+    - compute.googleapis.com
+    - dns.googleapis.com
+    - networkmanagement.googleapis.com
+    - networksecurity.googleapis.com
+    - servicenetworking.googleapis.com
+    - stackdriver.googleapis.com
+    - vpcaccess.googleapis.com
+  shared_vpc_host_config:
+    enabled: true
+  iam:
+    roles/owner:
+      - group:prj-admins@example.com
+  org_policies:
+    compute.restrictLoadBalancerCreationForTypes:
+      rules:
+        - allow:
+            values:
+              - EXTERNAL_HTTP_HTTPS
+# tftest-file id=project-config path=recipes/examples/project-config.yaml
```

Comment:
add schema to allow tests to validate the YAML (see [here](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/tree/master/modules/project#:~:text=policies/boolean.yaml-,schema,-%3Dorg%2Dpolicies.schema))

same for all other yaml examples